app/models/chargeback/rates_cache.rb

```diff
@@ -0,0 +1,24 @@
+class Chargeback
+  class RatesCache
+    def get(perf)
+      @rates ||= {}
+      @rates[perf.hash_features_affecting_rate] ||=
+        begin
+          prefix = tag_prefix(perf)
```

Comment:
Yes, it is the same. :+1: 

Each `Chargeback.descendant` has `where_clause` method. This `where_class` limits the resource_type of perf records coming here.